mantidproject/mantid
# Clean up output when running inelastic tests

### Description of work
This PR makes several changes to the inelastic tests to make it easier to read the output when running the tests:
- `NiceMock` is now used to hide "uninteresting mock" warnings
- A warning when creating test data without a source or position set is now avoided
- Algorithm logging when creating some test workspaces has been turned off
- The cxxtest suite framework now displays the name of the suite being run before its tests get run. This makes it easier to identify which test is causing an issue
- A warning about a QVBoxLayout not having a parent is now avoided

I have also separated the tests into separate folders for each interface so that it mirrors the parent directory.

Fixes #36579

### To test:
Run the `MantidQtInterfacesInelasticTestQt5` tests locally. The output should look something like this. Note that there is still a lot of logging for the DataManipulations tests, but this will be fixed when the tests are re-written in a future PR
```
Running 319 tests
class InelasticDataManipulationSymmetriseTabModelTestSymmetrise-[Notice] Symmetrise started
Symmetrise-[Notice] Symmetrise successful, Duration 0.00 seconds
.ScaleX-[Notice] ScaleX started
ScaleX-[Notice] ScaleX successful, Duration 0.00 seconds
SortXAxis-[Notice] SortXAxis started
SortXAxis-[Notice] SortXAxis successful, Duration 0.00 seconds
Symmetrise-[Notice] Symmetrise started
Symmetrise-[Notice] Symmetrise successful, Duration 0.00 seconds
.Symmetrise-[Notice] Symmetrise started
Symmetrise-[Notice] Symmetrise successful, Duration 0.00 seconds
.ScaleX-[Notice] ScaleX started
ScaleX-[Notice] ScaleX successful, Duration 0.00 seconds
SortXAxis-[Notice] SortXAxis started
SortXAxis-[Notice] SortXAxis successful, Duration 0.00 seconds
Symmetrise-[Notice] Symmetrise started
Symmetrise-[Notice] Symmetrise successful, Duration 0.00 seconds
.
class InelasticDataManipulationSqwTabModelTest..Rebin-[Notice] Rebin started
Rebin-[Notice] Rebin successful, Duration 0.03 seconds
.SofQW-[Notice] SofQW started
SofQW-[Notice] SofQW successful, Duration 0.02 seconds
AddSampleLog-[Notice] AddSampleLog started
AddSampleLog-[Notice] AddSampleLog successful, Duration 0.00 seconds
.Rebin-[Notice] Rebin started
Rebin-[Notice] Rebin successful, Duration 0.00 seconds
SofQW-[Notice] SofQW started
SofQW-[Notice] SofQW successful, Duration 0.02 seconds
AddSampleLog-[Notice] AddSampleLog started
AddSampleLog-[Notice] AddSampleLog successful, Duration 0.00 seconds
.
class InelasticDataManipulationMomentsTabModelTest..
class InelasticDataManipulationIqtTabModelTestTransformToIqt-[Notice] TransformToIqt started
TransformToIqt-[Notice] TransformToIqt successful, Duration 0.00 seconds
.
class InelasticDataManipulationElwinTabModelTestElasticWindowMultiple-[Notice] ElasticWindowMultiple started
ElasticWindowMultiple-[Notice] ElasticWindowMultiple successful, Duration 0.00 seconds
.GroupWorkspaces-[Notice] GroupWorkspaces started
GroupWorkspaces-[Notice] GroupWorkspaces successful, Duration 0.00 seconds
UnGroupWorkspace-[Notice] UnGroupWorkspace started
UnGroupWorkspace-[Notice] UnGroupWorkspace successful, Duration 0.00 seconds
.LoadNexus-[Notice] LoadNexus started
LoadNexusProcessed-[Warning] Error loading Instrument section of nxs file
LoadNexusProcessed-[Warning] Failed to find an instrument with this name in any facility: 'Unknown' - search object 'Unknown'
LoadNexusProcessed-[Warning] Try running LoadInstrument Algorithm on the Workspace to update the geometry
LoadNexus-[Notice] LoadNexus successful, Duration 0.11 seconds
.ExtractSingleSpectrum-[Notice] ExtractSingleSpectrum started
ExtractSingleSpectrum-[Notice] ExtractSingleSpectrum successful, Duration 0.01 seconds
ExtractSingleSpectrum-[Notice] ExtractSingleSpectrum started
ExtractSingleSpectrum-[Notice] ExtractSingleSpectrum successful, Duration 0.01 seconds
AppendSpectra-[Notice] AppendSpectra started
AppendSpectra-[Notice] AppendSpectra successful, Duration 0.00 seconds
.
class IndirectFittingModelTest.........................................
class IndirectFitPropertyBrowserTest..........................
class IndirectFitPlotPresenterTest....................................
class IndirectFitPlotModelTest.......................
class IndirectFitOutputOptionsPresenterTest.........................
class IndirectFitOutputOptionsModelTest..............................
class IndirectFitOutputTest........
class IndirectFitDataModelTest.........................
class IndirectFitDataTest......................................
class IndirectFitDataPresenterTest.............
class IndirectDataAnalysisTabTest....
class ConvFitDataPresenterTest.....
class FqFitDataPresenterTest........
class FqFitModelTest.
class IqtFitModelTest.
class IDAFunctionParameterEstimationTest..
class ConvFunctionModelTest............
class ConvFitModelTest.....OK!
```

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
